Mage.Sets/src/mage/cards/s/SanctumOfCalmWaters.java

```diff
@@ -0,0 +1,46 @@
+package mage.cards.s;
+
+import mage.abilities.Ability;
+import mage.abilities.common.BeginningOfPreCombatMainTriggeredAbility;
+import mage.abilities.dynamicvalue.common.PermanentsOnBattlefieldCount;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.abilities.effects.common.discard.DiscardControllerEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.constants.TargetController;
+import mage.filter.common.FilterControlledPermanent;
+
+import java.util.UUID;
+
+/**
+ * @author jmharmon
+ */
+
+public final class SanctumOfCalmWaters extends CardImpl {
+
+    public SanctumOfCalmWaters(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{3}{U}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.SHRINE);
+
+        // At the beginning of your precombat main phase, you may draw X cards, where X is the number of Shrines you control. If you do, discard a card.
+        Ability ability = new BeginningOfPreCombatMainTriggeredAbility(new DrawCardSourceControllerEffect(new PermanentsOnBattlefieldCount(new FilterControlledPermanent(SubType.SHRINE)))
+                .setText("At the beginning of your precombat main phase, you may draw X cards, where X is the number of Shrines you control"),
```

Comment:
This text should not include "At the beginning of your precombat main phase, ". That text already comes from the ability. I fixed it in my PR [here](https://github.com/magefree/mage/pull/6708).